frame/alliance/src/lib.rs

```diff
@@ -0,0 +1,1049 @@
+
+	/// Maps member type to members of each type.
+	#[pallet::storage]
+	#[pallet::getter(fn members)]
+	pub type Members<T: Config<I>, I: 'static = ()> = StorageMap<
+		_,
+		Twox64Concat,
+		MemberRole,
+		BoundedVec<T::AccountId, T::MaxMembersCount>,
+		ValueQuery,
+	>;
+
+	/// A set of members that are (potentially) being kicked out. They cannot retire until the
+	/// motion is settled.
+	#[pallet::storage]
+	#[pallet::getter(fn up_for_kicking)]
+	pub type UpForKicking<T: Config<I>, I: 'static = ()> =
+		StorageMap<_, Blake2_128Concat, T::AccountId, bool, ValueQuery>;
+
+	/// The current blacklist of accounts. These accounts cannot submit candidacy.
+	#[pallet::storage]
+	#[pallet::getter(fn account_blacklist)]
+	pub type AccountBlacklist<T: Config<I>, I: 'static = ()> =
+		StorageValue<_, BoundedVec<T::AccountId, T::MaxBlacklistCount>, ValueQuery>;
+
+	/// The current blacklist of websites.
+	#[pallet::storage]
+	#[pallet::getter(fn website_blacklist)]
+	pub type WebsiteBlacklist<T: Config<I>, I: 'static = ()> =
+		StorageValue<_, BoundedVec<UrlOf<T, I>, T::MaxBlacklistCount>, ValueQuery>;
+
+	#[pallet::call]
+	impl<T: Config<I>, I: 'static> Pallet<T, I> {
+		/// Add a new proposal to be voted on.
+		///
+		/// Requires the sender to be a founder or fellow.
+		#[pallet::weight((
+			T::WeightInfo::propose_proposed(
+				*length_bound, // B
+				T::MaxFounders::get(), // X
+				T::MaxFellows::get(), // Y
+				T::MaxProposals::get(), // P2
+			),
+			DispatchClass::Normal
+		))]
+		pub fn propose(
+			origin: OriginFor<T>,
+			#[pallet::compact] threshold: u32,
+			proposal: Box<<T as Config<I>>::Proposal>,
+			#[pallet::compact] length_bound: u32,
+		) -> DispatchResult {
+			let proposor = ensure_signed(origin)?;
+			ensure!(Self::has_voting_rights(&proposor), Error::<T, I>::NoVotingRights);
+
+			if let Some(Call::kick_member { who }) = proposal.is_sub_type() {
+				let strike = T::Lookup::lookup(who.clone())?;
+				<UpForKicking<T, I>>::insert(strike, true);
+			}
+
+			T::ProposalProvider::propose_proposal(proposor, threshold, proposal, length_bound)?;
+			Ok(())
+		}
+
+		/// Add an aye or nay vote for the sender to the given proposal.
+		///
+		/// Requires the sender to be a founder or fellow.
+		#[pallet::weight((
+			T::WeightInfo::vote(T::MaxFounders::get(), T::MaxFellows::get()),
+			DispatchClass::Operational
+		))]
+		pub fn vote(
+			origin: OriginFor<T>,
+			proposal: T::Hash,
+			#[pallet::compact] index: ProposalIndex,
+			approve: bool,
+		) -> DispatchResult {
+			let who = ensure_signed(origin)?;
+			ensure!(Self::has_voting_rights(&who), Error::<T, I>::NoVotingRights);
+
+			T::ProposalProvider::vote_proposal(who, proposal, index, approve)?;
+			Ok(())
+		}
+
+		/// Veto a proposal about `set_rule` and `elevate_ally`, close, and remove it from the
+		/// system, regardless of its current state.
+		///
+		/// Must be called by a founder.
+		#[pallet::weight(T::WeightInfo::veto(T::MaxProposals::get()))]
```

Comment:
```suggestion
		#[pallet::weight((
		    T::WeightInfo::veto(T::MaxProposals::get()),
		    DispatchClass::Operational
		))]
```
if vote is important enough to be operational dispatch class then veto should be too?